docs/source/docker/advanced/datasource_predictions.rst

```diff
@@ -0,0 +1,347 @@
+Here, `file_name` serves as a unique identifier to retrieve the image for which
+the predictions are made and predictions is a list of `Prediction Singletons` for the corresponding task.
+
+Example classification:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/classification_weather/my_image.json
+
+    {
+        "file_name": "my_image.png"
+        "predictions": [ // classes: [sunny, clouded, rainy]
+            {
+                "category_id": 0,
+                "probabilities": [0.8, 0.1, 0.1]
+            }
+        ]
+    }
+
+Example object detection:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/object_detection/my_image.json
+
+    {
+        "file_name": "my_image.png",
+        "predictions": [ // classes: [person, car]
+            {
+                "category_id": 0,
+                "bbox": [...],
+                "score": 0.8
+            },
+            {
+                "category_id": 1,
+                "bbox": [...],
+                "score": 0.9
+            },
+            {
+                "category_id": 0,
+                "bbox": [...],
+                "score": 0.5
+            }
+        ]
+    }
+
+Note: The filename should always be the full path from the root directory.
+
+
+Prediction Singletons
+---------------------
+The prediction singletons closely follow the `COCO results <https://cocodataset.org/#format-results>`_ format while dropping
+the `image_id`. Note the the `category_id` must be the same as the one defined
+in the schema and that the probabilities (if provided) must follow the order of the category ids.
+
+**Classification:**
+
+For classification, please use the following format:
+
+.. code-block:: javascript
+
+    [{
+        "category_id"       : int,
+        "probabilities"     : [p0, p1, ..., pN],
+    }]
+
+**Object Detection:**
+
+For detection with bounding boxes, please use the following format:
+
+.. code-block:: javascript
+
+    [{
+        "category_id"       : int,
+        "bbox"              : [x, y, width, height],
+        "score"             : float,
+        "probabilities"     : [p0, p1, ..., pN],         // optional
+    }]
+
+The bounding box format follows the `COCO results <https://cocodataset.org/#format-results>`_ documentation.
+
+.. note::
+
+    Box coordinates are floats measured from the top left image corner (and are 0-indexed).
+    We recommend rounding coordinates to the nearest tenth of a pixel to reduce resulting JSON file size.
+
+
+Creating the predictions folder
+-------------------------------
+
+For creating the predictions folder, we recommend writing a script that takes your predictions and
+saves them in the format just outlined. You can either save the predictions first on your local machine
+and then upload them for your datasource or upload them directly to your datasource.
+
+As an example, the following script takes an object detection annotations file in the coco format.
```

Comment:
It's great to provide a script. I'd maybe add an example or link to an example of how the format looks like. 
Furthermore, we provide a script to turn annotations into predictions? That seems rather like a hack and not what I'd suggest every user to do. We also set the score for every file to 0 and we explain in the other section that we will do active learning using the predictions one would obtain from this guide :S